controller/execute.go

```diff
@@ -0,0 +1,457 @@
+		p, err = scaleway.NewScalewayProvider(ctx, domainFilter, cfg.DryRun)
+	case "godaddy":
+		p, err = godaddy.NewGoDaddyProvider(ctx, domainFilter, cfg.GoDaddyTTL, cfg.GoDaddyAPIKey, cfg.GoDaddySecretKey, cfg.GoDaddyOTE, cfg.DryRun)
+	case "gandi":
+		p, err = gandi.NewGandiProvider(ctx, domainFilter, cfg.DryRun)
+	case "pihole":
+		p, err = pihole.NewPiholeProvider(
+			pihole.PiholeConfig{
+				Server:                cfg.PiholeServer,
+				Password:              cfg.PiholePassword,
+				TLSInsecureSkipVerify: cfg.PiholeTLSInsecureSkipVerify,
+				DomainFilter:          domainFilter,
+				DryRun:                cfg.DryRun,
+			},
+		)
+	case "ibmcloud":
+		p, err = ibmcloud.NewIBMCloudProvider(cfg.IBMCloudConfigFile, domainFilter, zoneIDFilter, endpointsSource, cfg.IBMCloudProxied, cfg.DryRun)
+	case "plural":
+		p, err = plural.NewPluralProvider(cfg.PluralCluster, cfg.PluralProvider)
+	case "tencentcloud":
+		p, err = tencentcloud.NewTencentCloudProvider(domainFilter, zoneIDFilter, cfg.TencentCloudConfigFile, cfg.TencentCloudZoneType, cfg.DryRun)
+	case "webhook":
+		p, err = webhook.NewWebhookProvider(cfg.WebhookProviderURL)
+	default:
+		log.Fatalf("unknown dns provider: %s", cfg.Provider)
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if cfg.WebhookServer {
+		webhookapi.StartHTTPApi(p, nil, cfg.WebhookProviderReadTimeout, cfg.WebhookProviderWriteTimeout, "127.0.0.1:8888")
+		os.Exit(0)
+	}
+
+	if cfg.ProviderCacheTime > 0 {
+		p = provider.NewCachedProvider(
+			p,
+			cfg.ProviderCacheTime,
+		)
+	}
+
+	reg, err := selectRegistry(cfg, p)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	policy, exists := plan.Policies[cfg.Policy]
+	if !exists {
+		log.Fatalf("unknown policy: %s", cfg.Policy)
+	}
+
+	ctrl := Controller{
+		Source:               endpointsSource,
+		Registry:             reg,
+		Policy:               policy,
+		Interval:             cfg.Interval,
+		DomainFilter:         domainFilter,
+		ManagedRecordTypes:   cfg.ManagedDNSRecordTypes,
+		ExcludeRecordTypes:   cfg.ExcludeDNSRecordTypes,
+		MinEventSyncInterval: cfg.MinEventSyncInterval,
+	}
+
+	if cfg.Once {
+		err := ctrl.RunOnce(ctx)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		os.Exit(0)
+	}
+
+	if cfg.UpdateEvents {
+		// Add RunOnce as the handler function that will be called when ingress/service sources have changed.
+		// Note that k8s Informers will perform an initial list operation, which results in the handler
+		// function initially being called for every Service/Ingress that exists
+		ctrl.Source.AddEventHandler(ctx, func() { ctrl.ScheduleRunOnce(time.Now()) })
+	}
+
+	ctrl.ScheduleRunOnce(time.Now())
+	ctrl.Run(ctx)
+}
+
+// This function configures the logger format and level based on the provided configuration.
+func configureLogger(cfg *externaldns.Config) {
+	logFormats := []string{"text", "json"}
+	if !slices.Contains(logFormats, cfg.LogFormat) {
+		log.Fatalf("unknown log format: '%s'. known formats: '%s'", cfg.LogFormat, strings.Join(logFormats, ","))
+	}
```

Comment:
Wouldn't the log format check be a bit redundant?
There's already:

https://github.com/conduitxyz/external-dns/blob/e64e5368eeca63345d5500680d9282fc579312a7/pkg/apis/externaldns/types.go#L619
and
https://github.com/conduitxyz/external-dns/blob/e64e5368eeca63345d5500680d9282fc579312a7/pkg/apis/externaldns/validation/validation.go#L31-L33

that should catch the wrong format sooner?

Reply:
![Screenshot 2025-03-29 at 18 49 36](https://github.com/user-attachments/assets/0815e7a3-73ac-4caf-b1c8-c2bc3cd16d32)

removed